### PR TITLE
refactor: address namespace follow-ups from celestia-app v1.0.0-rc12 bump

### PR DESCRIPTION
**Pull request by rootulp**
Addresses two follow-ups from https://github.com/celestiaorg/celestia-node/pull/2581
Closes https://github.com/celestiaorg/celestia-node/issues/2561

### CODE REVIEW

File: share/namespace.go
Comment (Wondertan) [inline share/namespace.go:133]: Let's avoid conversions and copy code for now.

**Review by Wondertan — COMMENTED** [review]
Thank you!

File: share/namespace.go
Comment (rootulp) [reply share/namespace.go]: It seems like a burden to maintain the same validation logic in two separate repos but defer to maintainers.

Resolved via f0f02f1a47e67f0c669291a775730fdf8750ef05

**Review by Wondertan — APPROVED** [review]

**Pull request merged**
